src/sentry/web/api.py

```diff
@@ -332,19 +348,29 @@ def process(self, request, project, key, auth, helper, data, **kwargs):
             sender=type(self),
         )
 
-        should_filter = helper.should_filter(project, data, ip_address=remote_addr)
+        should_filter = helper.should_filter(
+            project, data, ip_address=remote_addr)
         if should_filter[0]:
             tsdb.incr_multi(
                 [
                     (tsdb.models.project_total_received, project.id),
                     (tsdb.models.project_total_blacklisted, project.id),
-                    (tsdb.models.organization_total_received, project.organization_id),
-                    (tsdb.models.organization_total_blacklisted, project.organization_id),
+                    (tsdb.models.organization_total_received,
+                     project.organization_id),
+                    (tsdb.models.organization_total_blacklisted,
+                     project.organization_id),
                     (tsdb.models.key_total_received, key.id),
                     (tsdb.models.key_total_blacklisted, key.id),
                 ]
             )
-            metrics.incr('events.blacklisted', tags={'reason': should_filter[1]})
+            try:
+                tsdb.incr(
+                    FILTER_STAT_KEYS_TO_VALUES[should_filter[1]], project.id)
+            except KeyError:
```

Comment:
Ideally this `try` block would be limited to the `FILTER_STAT_KEYS_TO_VALUES[should_filter[1]]` expression to prevent accidentally catching `KeyError`s that could be raised during `tsdb.incr`. (Though when paired with the above comment this shouldn't be a concern anyway.)

Reply:
I'm not really sure how to do the try/except on KeyError while it's part of the`incr_multi` list - is that possible?

Reply:
Nope, what you did in cf77ef9cfc471c04711e55585c8aa8b4ee4d9d96 is good though.